### PR TITLE
refactor: Move CALL_STATE into best_practices

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Move CALL_STATE enum and associated checks outside of core_validation
and into best_practices.

Change-Id: Id842fe45f32feae73333f5bfbc20d6cf10878d02

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,5 @@ build-android/third_party
 *.includes
 .vscode/
 .DS_Store
+
+.*.sw[op]
```
Comment (mark-lunarg) [inline .gitignore:38]: This should prolly live in a separate commit.

File: layers/best_practices_validation.h [hunk 2]
```diff
@@ -53,6 +53,28 @@ typedef enum {
 } BPVendorFlagBits;
 typedef VkFlags BPVendorFlags;
 
+enum CALL_STATE {
+    UNCALLED,       // Function has not been called
+    QUERY_COUNT,    // Function called once to query a count
+    QUERY_DETAILS,  // Function called w/ a count to query details
+};
+
```
Comment (locke-lunarg) [inline layers/best_practices_validation.h:61]: In this PR, https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2199/files

You also moved  `CALL_STATE` into `state_tracker.h`. Maybe you could merge these two branches into one and leave one PR.

Comment (ncesario-lunarg) [reply layers/best_practices_validation.h:61]: Yep, going to close #2199 in favor of this change.

File: layers/best_practices_validation.h [hunk 3]
```diff
@@ -304,4 +384,20 @@ class BestPractices : public ValidationStateTracker {
 
     // used to track depth pre-pass heuristic data per command buffer
     std::unordered_map<VkCommandBuffer, DepthPrePassState> cbDepthPrePassStates = {};
+
+    // Used for instance versions of this object
+    std::unordered_map<VkSwapchainKHR, SWAPCHAIN_STATE_BP> swapchain_state_map;
```
Comment (locke-lunarg) [inline layers/best_practices_validation.h:389]: I think swapchain_bp_state_map is more clear.

File: layers/best_practices_utils.cpp [hunk 4]
```diff
@@ -175,8 +175,8 @@ bool BestPractices::PreCallValidateCreateDevice(VkPhysicalDevice physicalDevice,
                                              kVUID_BestPractices_CreateDevice_DeprecatedExtension);
     }
 
-    auto pd_state = GetPhysicalDeviceState(physicalDevice);
-    if ((pd_state->vkGetPhysicalDeviceFeaturesState == UNCALLED) && (pCreateInfo->pEnabledFeatures != NULL)) {
+    const auto& bp_pd_state = phys_device_bp_state_map.at(physicalDevice);
```
Comment (mark-lunarg) [inline layers/best_practices_utils.cpp:178]: ELI5:  at() instead of find()?

Comment (mark-lunarg) [reply layers/best_practices_utils.cpp:178]: Or even GetPhysicalDeviceStateBP?

**Review by mark-lunarg — APPROVED** [review]
LGTM!

File: layers/best_practices_utils.cpp (hunk 4 above)
Comment (ncesario-lunarg) [reply layers/best_practices_utils.cpp:178]: Yep, think I just forgot to use GetPhysicalDeviceStateBP here. Thanks!
